x/mongo/driver/batch_cursor.go

```diff
@@ -94,6 +99,27 @@ func NewCursorResponse(response bsoncore.Document, server Server, desc descripti
 			}
 		}
 	}
+
+	// If the deployment is behind a load balancer and the cursor has a non-zero ID, pin the cursor to a connection and
+	// use the same connection to execute getMore and killCursors commands.
+	if curresp.Desc.ServerID != nil && curresp.ID != 0 {
+		// Cache the server as an ErrorProcessor to use when constructing deployments for cursor commands.
+		ep, ok := curresp.Server.(ErrorProcessor)
+		if !ok {
+			return CursorResponse{}, fmt.Errorf("expected Server used to establish a cursor to implement ErrorProcessor, but got %T", curresp.Server)
+		}
+		curresp.ErrorProcessor = ep
+
+		refConn, ok := info.Connection.(PinnedConnection)
```

Comment:
Is there a reason we need the `Connection` implementation itself to support pinning? If we were to create a wrapper around a `Connection` that did the same thing, would that work?

Reply:
Ah, got it.